packages/core/src/hooks/hookRegistry.ts

```diff
@@ -110,8 +110,12 @@ export class HookRegistry {
   private processHooksFromConfig(): void {
     // Get hooks from the main config (this comes from the merged settings)
     const configHooks = this.config.getHooks();
-    if (configHooks) {
+    if (configHooks && this.config.isTrustedFolder()) {
```

Comment:
The current logic in processHooksFromConfig will log a warning even when no hooks are configured. Nest the trust check to ensure we only warn when hooks are actually present but blocked.

```typescript
     const configHooks = this.config.getHooks();
     if (configHooks) {
       if (this.config.isTrustedFolder()) {
         this.processHooksConfiguration(configHooks, ConfigSource.Project);
       } else {
         debugLogger.warn(
           'Project hooks disabled because the folder is not trusted.',
         );
       }
     }
```